### PR TITLE
Completable job

### PR DESCRIPTION
**Pull request by elizarov**

### CODE REVIEW

File: kotlinx-coroutines-core/common/src/CompletableJob.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
```
Comment (qwwdfsad) [inline kotlinx-coroutines-core/common/src/CompletableJob.kt]: Please update copyright year

File: kotlinx-coroutines-core/common/src/CompletableJob.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     *
+     * Repeated invocations of this function have no effect and always produce `false`.
+     *
+     * Note, that if this job has children, then it transitions into _completing_ state and becomes _complete_
+     * once all its children are _complete_. See [Job] for details.
+     */
+    public fun complete(): Boolean
+
+    /**
+     * Completes this job exceptionally with a given [exception]. The result is `true` if this job was
```
Comment (qwwdfsad) [inline kotlinx-coroutines-core/common/src/CompletableJob.kt]: Please describe cancellation semantics of `completeExceptionally` here and in `CompletableDeferred`.

Comment (qwwdfsad) [reply kotlinx-coroutines-core/common/src/CompletableJob.kt]: Also, we can increase deprecation level of `Job.cancel(cause)`

Comment (elizarov) [reply kotlinx-coroutines-core/common/src/CompletableJob.kt]: We cannot simply increase deprecation level of `Job.cancel(cause)`, see #975

**Pull request merged**